content/articles/how-to-build-discord-bot-with-javascript/index.md

```diff
@@ -0,0 +1,163 @@
+Select a new terminal and type the following.
+```
+npm init -y
+```
+The code generates npm project without going through the interactive process
+In my case, the following are the result. Remove the script code.
+```
+{
+  "name": "mybot",
+  "version": "1.0.0",
+  "description": "",
+  "main": "bot.js",
+  "scripts": {
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC"
+}
+
+```
+```
+npm install discord.js dotenv
+```
+discord.js will allow you to interact with Discord API.
+dotenv will load variables from .env files.
+```
+npm install -D nodemon
+```
+nodemon restart the server every time we make a change to our files.
+### Building our bot
+First, we need to copy the bot token to allow calls in Discord.js:
+It's done by clicking the copy button in the bot tab. As shown below
+![](/education-engineering/how-to-build-discord-bot-with-javascript/Token.png)
+Once copied, open a new file and name it ``.env``, then paste it in the following format
+```
+TOKEN=Paste_token_here
+```
+### Create the bot file
+In my case, I will name my file as ``bot.js`` in the main project directory.
+First, we will set up environment variables using the ``dotenv`` package, import classes from ``discord.js``, and then start up a new client. This is accomplished utilizing the accompanying code:
+```
+require("dotenv").config(); //to start process from .env file
+const {Client, Intents}=require("discord.js");
+const client=new Client({
+    Intents:[
+        Intents.FLAGS.GUILDS,//adds server functionality
+        Intents.FLAGS.GUILD_MESSAGES //gets messages from our bot.
+    ]
+});
+client.once("ready", () =>{
+    console.log("BOT IS ONLINE");
+})
+client.login(process.env.TOKEN);
+```
+To run your bot, type the following in your terminal:
+```
+node bot.js
+```
+![](/education-engineering/how-to-build-discord-bot-with-javascript/botonline.png)
+As you can see above, our bot is online but it does not have any functionality.
+Note we are authenticating the named variable from our .env file, which authenticates our bot with Discord API. To test our bot we have to be logged into a Discord account and have a server running.
+### Have bot reply to messages
+So far our bot is online but has no active functionality, The following lines of code will have our bot reply to a simple hello message.
+```
+client.on('message',
+function (messages){
+    if(messages.content.toLocaleLowerCase()==='hello')
+    messages.channel.send('hello' + ' '  + messages.author.username);
+})
+```
+Our bot will reply hello message and mention the sender.
+![](/education-engineering/how-to-build-discord-bot-with-javascript/reply.png)
+### Summary
+Here is what we have covered:
+1. Add New Discord Application
+2. Create New Node.js Project
+3. Install Dependencies
+4. Create the bot file
+5. Build our bot
+6. Have bot reply to text
+Below is our entire code:

```

Comment:
Avoid bulleted points. Explain these in a paragraph with sentences related to each other.